### DISCUSSION TITLE
Question on how to work with ChatSession and KernelMemory together

### DISCUSSION THREAD

**Discussion by nicholusi2021** [SciSharp/LLamaSharp#792] [Q&A]
Is it possible to integrate IKernelMemory memory into a ChatSession object?

I have created and IKernelMemory object, loaded some pdf files and ask some questions via AskAsync, and that works great.
I have created a ChatSession object, and ask some questions via ChatAsync, and that works great.

I can use these two objects independently, but I haven't figured out how to combine them.  What I would like to do is have a conversation using a ChatSession so that the ChatHistory is used for future answers in the session, but also use KernelMemory to load some PDF files for ChatSession to utilize when it creates the answer.  Is this possible?  Thanks!

**Comment by SanftMonster** [comment]
I think it's possible but requires some new features. Do you mean that you want the chat session do the RAG search every loop of chat?

**Reply by nicholusi2021** [reply]
Yes, I would like to incorporate custom data into the memory and every time a question is asked in the ChatSession, it references that data.

**Reply by dameat2004** [reply]
Im having similar issue. Im trying to use SemanticKernel with KernelMemory that stores data from files in sql, and hopefully add chatcompletion with chathistory all working together. Got stuck with semanticKernel and KernelMemory plugin not working together for some reason.

**Reply by dameat2004** [reply]
forgot, here is some similar example just not using llamasharp [https://www.developerscantina.com/p/semantic-kernel-memory
]

**Reply by biapar** [reply]
I've the same problem. Have you solved?
